File: onebot-sdk/src/main/java/cn/evole/onebot/sdk/event/notice/NoticeEvent.java
package cn.evole.onebot.sdk.event.notice;

import cn.evole.onebot.sdk.event.Event;
import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

/**
 * Created on 2022/7/8.
 *
 * @author cnlimiter
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
@EqualsAndHashCode(callSuper = true)
public class NoticeEvent extends Event {

    @SerializedName( "notice_type")
    private String noticeType;

    @SerializedName( "user_id")
    private long userId;

}

File: onebot/src/main/kotlin/sdk/event/notice/NoticeEvent.kt
package cn.evolvefield.onebot.sdk.event.notice

import cn.evolvefield.onebot.sdk.event.Event
import com.google.gson.annotations.SerializedName
import lombok.AllArgsConstructor
import lombok.Data
import lombok.EqualsAndHashCode
import lombok.NoArgsConstructor
import lombok.experimental.SuperBuilder

@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
@EqualsAndHashCode(callSuper = true)
open class NoticeEvent : Event() {
    @SerializedName("notice_type")
    var noticeType = ""
    @SerializedName("user_id")
    var userId = 0L
}
